testgrid/web/src/components/views/InstanceTable.jsx

```diff
@@ -14,29 +15,60 @@ import Loader from "../shared/Loader";
 import "../../assets/scss/components/InstanceTable.scss";
 
 import { getNodeLogs } from "./dataFetchers";
-export default class InstanceTable extends React.Component {
-  constructor(props) {
-    super(props);
-
-    this.state = {
-      showInstallerModal: false,
-      showLogsModal: false,
-      selectedInstance: null,
-      selectedNode: null,
-      instanceLogs: "",
-      loadingLogs: false,
-      sonobuoyResults: "",
-      showSonobuoyResultsModal: false,
-      loadingSonobuoyResults: false,
-      activeMarkers: [],
-      showUpgradeYaml: false,
-      showSupportbundleYaml: false,
-      showPostInstallScript: false,
-      showPostUpgradeScript: false,
-    };
+
+const InstanceTable = (props) => {
+  const [searchParams] = useSearchParams();
+  const location = useLocation();
+  const navigate = useNavigate();
+
+  const [selectedInstance, setSelectedInstance] = useState(null);
+  const [selectedNode, setSelectedNode] = useState(null);
+  const [instanceLogs, setInstanceLogs] = useState("");
+  const [sonobuoyResults, setSonobuoyResults] = useState("");
+  const [activeMarkers, setActiveMarkers] = useState([]);
+  const [showInstallerModal, setShowInstallerModal] = useState(false);
+  const [showLogsModal, setShowLogsModal] = useState(false);
+  const [loadingLogs, setLoadingLogs] = useState(false);
+  const [showSonobuoyResultsModal, setShowSonobuoyResultsModal] = useState(false);
+  const [loadingSonobuoyResults, setLoadingSonobuoyResults] = useState(false);
+  const [showUpgradeYaml, setShowUpgradeYaml] = useState(false);
+  const [showSupportbundleYaml, setShowSupportbundleYaml] = useState(false);
+  const [showPostInstallScript, setShowPostInstallScript] = useState(false);
+  const [showPostUpgradeScript, setShowPostUpgradeScript] = useState(false);
+
+  // TODO: this does not work
```

Comment:
this still does not work but i'd like to merge this as it makes things usable again